src/messages/MClientReply.h

```diff
@@ -390,13 +386,26 @@ class MClientReply final : public MMDSOp {
     using ceph::decode;
     auto p = payload.cbegin();
     decode(head, p);
+    // errorcode32_t implements conversion from/to different host error codes
+    // casting needed since error codes are signed int32 and head.result is unsigned int32
+    // errortype_t::code_t is alias for __s32, which is signed
+    // ceph_mds_reply_head::code_t is alias for __le32 which is unsigned
+    errorcode32_t temp{static_cast<errorcode32_t::code_t>(head.result)};
+    temp.set_wire_to_host();
+    head.result = static_cast<ceph_mds_reply_head::code_t>(temp.code);
```

Comment:
So this is a bit weird. The integers set in `head.result` are probably negative because the MDS was using `-CEPHFS_E*` and the `CEPHFS_E` codes were all positive. However, `head.result` is an unsigned.

But none of that should matter because the `ceph_to_hostos` routines all discard the sign when converting the errno. Therefore, I think it's safe to change ` struct ceph_mds_reply_head::result` type to `errorcode32_t`. (Please add a comment to the struct that `errorcode32_t` is not a primitive type and has special encode/decode routines, but that it is treated like a signed int32_t. This will be important for keeping in sync with the linux kernel.) 

Also note: the Linux kernel is casting the `__le32` to an `int`.